dev/com.ibm.ws.logging.osgi/src/com/ibm/ws/logging/internal/osgi/stackjoiner/MethodProxy.java

```diff
@@ -0,0 +1,68 @@
+/*******************************************************************************
+ * Copyright (c) 2021 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.logging.internal.osgi.stackjoiner;
+
+import java.io.PrintStream;
+import java.lang.instrument.Instrumentation;
+import java.lang.reflect.Method;
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+
+public class MethodProxy {
+	
+	private static final TraceComponent tc = Tr.register(MethodProxy.class);
+	private Method method;
+	private String className;
+	private String methodName;
+	
+	public MethodProxy(Instrumentation inst, String className, String methodName) {
+		this.className = className;
+		this.methodName = methodName;
+		Class<?> classRetrieved = retrieveClass(inst, className);
+		if (classRetrieved != null) {
+			Method method = ReflectionHelper.getDeclaredMethod(classRetrieved, methodName, Throwable.class, PrintStream.class);
```

Comment:
need to move the Throwable.class and PrintStream.class to the caller as well (otherwise this still is overly tied to the one kind of method it can proxy)

Reply:
Fixed in [387e245](https://github.com/OpenLiberty/open-liberty/pull/16276/commits/387e2458222ec489860c8281376aecc6d4bf1687.